Examples/Scripts/MaterialMapping/materialComposition.C

```diff
@@ -58,14 +60,16 @@ struct MaterialHistograms {
                              M_PI, 0., 5.);
   }
 
-  MaterialHistograms() = default;
-
+  // yes we might leak memory here but this would break copy (and potentially
+  // move) and I dont want to implement them
+  /*
   ~MaterialHistograms() {
     delete x0_vs_eta;
     delete l0_vs_eta;
     delete x0_vs_phi;
     delete l0_vs_phi;
   }
+  */
```

Comment:
Could you give an estimate an estimate on how much memory we might be losing?

And maybe we should also add this as a warning to the PR-description (and open an issue) - or better: find someone who wants to implement them.

Reply:
simple solution would be to make them to `shared_ptr` (or `unique_ptr` if copy is not needed). That gives you everything for free.

Reply:
but do you actually do that with root objects?

also: paul mentioned that the profiles might be owned by the canvas?

Reply:
Actually its worth to read once through https://root.cern/manual/object_ownership/

And exactly, all histograms are owned by default by the current `gDirectory` object of root. Can be disabled, but in our case it should be just fine to remove the constructor.

Reply:
I actually would remove the comment and the destructor completely, since there should be no memory leak.